### PR TITLE
modify train_rl to use mean_return as well as success_rate

### PR DESCRIPTION
**Pull request by dyth**

### CODE REVIEW

**Comment by maximecb** [comment]
Hmm, so I guess that's more intuitive to someone watching the logs, but RL is optimizing the returns, not directly the success rate.

**Comment by dyth** [comment]
Originally the script was optimising for the success rate.
The statements I added would optimise for the return if the success rate is the same.
Once the success rate has reached 100 %, no more weights would be saved in the current script, even though the return value may still increase.
The modification should prevent this from happening.

**Comment by maximecb** [comment]
@rizar Do you have an opinion on this?

**Comment by rizar** [comment]
It makes to resave the best model if the success rate is still good but the return is better. This can be achieved with less copy-paste, though ;)

File: scripts/train_rl.py [hunk 1]
```diff
@@ -238,5 +239,10 @@
             utils.save_model(acmodel, args.model + '_best')
             obss_preprocessor.vocab.save(utils.get_vocab_path(args.model + '_best'))
             logger.info("Return {: .2f}; best model is saved".format(mean_return))
+        if (success_rate == best_success_rate) and (mean_return > best_mean_return):
+            best_mean_return = mean_return
```
Comment (rizar) [inline scripts/train_rl.py]: Would be nicer to have just one `if` block here.

**Comment by maximecb** [comment]
Sorry about the failing checks. That's my doing. Submitted a fix: https://github.com/mila-iqia/babyai/pull/61

**Pull request merged**